site/src/pages/RoadmapPage/sidebar/ProgramRequirementsList.tsx

```diff
@@ -88,15 +88,31 @@ const CourseTile: FC<CourseTileProps> = ({ courseID, dragTimestamp = 0, taken })
     fontSize = computedSize + 'px';
   }
 
+  const sourceOverlay =
+    completedBy === 'AP' ? (
+      <p className="source-overlay" title={`${courseID} is cleared by an AP exam`}>
+        AP
+      </p>
+    ) : completedBy === 'Course' ? (
+      <p className="source-overlay" title={`${courseID} is cleared by a transferred course`}>
+        Course {/* to do: replace with MUI icon */}
+      </p>
+    ) : null;
+
   return (
     <div className={className} {...tappableCourseProps} style={{ fontSize }}>
+      {sourceOverlay}
```

Comment:
Then simply call the component here:
```suggestion
      <SourceOverlay completedBy={completedBy} />
```